modules/claims_api/app/workers/claims_api/ews_updater.rb

```diff
@@ -78,14 +80,19 @@ def update_bgs_claim(ews, bgs_claim)
         ews.status = ClaimsApi::EvidenceWaiverSubmission::ERRORED
         ews.bgs_error_message = "BGS Error: update_record failed with code #{response[:return_code]}"
         ews.bgs_upload_failure_count = ews.bgs_upload_failure_count + 1
-        ClaimsApi::Logger.log(ews_id: ews.id, claim_id: ews.claim_id,
-                              detail: 'Waiver update Failed', error: response[:return_code],
-                              failure_count: ews.bgs_upload_failure_count)
+        ClaimsApi::Logger.log('ews_updater',
+                              detail: "Failed to waiver update claim: #{ews.claim_id},
+                              and ews: #{ews.id}, with return code: #{response[:return_code]},
+                              and failure count: #{ews.bgs_upload_failure_count}.")
       else
-        ews.status = ClaimsApi::EvidenceWaiverSubmission::UPDATED
+        update_claim_level_suspense(ews)
         # Clear out the error message if there were previous failures
         ews.bgs_error_message = nil if ews.bgs_error_message.present?
-        ClaimsApi::Logger.log({ ews_id: ews.id, claim_id: ews.claim_id, detail: 'Waiver Success' })
+        ews.status = ClaimsApi::EvidenceWaiverSubmission::UPDATED
+        ews.save
+        ClaimsApi::Logger.log('ews_updater',
+                              detail: "Successfully updated the waiver for
+                              claim: #{ews.claim_id}, and ews: #{ews.id}.")
```

Comment:
this flow returns the same issue we're seeing now (that I introduced) where we're setting `errored` status and saving it in the `update_claim_level_suspense` rescue, but the job will then continue and sets the `updated` status....